src/zarr/api/asynchronous.py

```diff
@@ -1120,8 +1119,10 @@ async def empty_like(
     retrieve data from an empty Zarr array, any values may be returned,
     and these are not guaranteed to be stable from one access to the next.
     """
-    like_kwargs = _like_args(a, kwargs)
-    return await empty(**like_kwargs)
+    like_kwargs = _like_args(a) | kwargs
+    if isinstance(a, (AsyncArray | Array)):
+        like_kwargs.setdefault("fill_value", a.metadata.fill_value)
```

Comment:
To make sure I understand the precedence, it's

1. User keyword arguments (via `| kwargs`)
2. `_like_kwargs(a)`
3. `fill_value` via `a.metadata.fill_value`

? It'd be worth confirming we have each of those cases covered with tests.

Reply:
the first two cases were handled already, and I added fill_value testing in https://github.com/zarr-developers/zarr-python/pull/2992/commits/d29f68275505596b7ac14c1445b61a68c748d142